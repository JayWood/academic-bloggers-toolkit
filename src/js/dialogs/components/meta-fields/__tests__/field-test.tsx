import * as React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import ManualDataStore from 'stores/data/manual-data-store';
import Field from '../field';

const setup = () => {
    const meta = new ManualDataStore('webpage', '12345');
    const field = {
        label: 'Title',
        value: 'title',
    };
    const component = shallow(
        <Field field={field} meta={meta} onChange={jest.fn()} />,
    );
    return {
        component,
        meta,
    };
};

describe('<Field />', () => {
    it('should match snapshots', () => {
        const { component, meta } = setup();
        const before = toJSON(component);
        expect(before).toMatchSnapshot();

        meta.updateField('title', 'TEST');
        component.update();
        const after = toJSON(component);
        expect(after).toMatchDiffSnapshot(before);
    });
});
