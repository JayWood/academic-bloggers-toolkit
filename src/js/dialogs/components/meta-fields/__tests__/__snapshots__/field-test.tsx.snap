// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Field /> should match snapshots 1`] = `
<div
  className="row"
>
  <label
    className="label"
    htmlFor="field-title"
  >
    Title
  </label>
  <input
    className="input"
    data-field="title"
    id="field-title"
    onChange={[MockFunction]}
    type="text"
    value=""
  />
</div>
`;

exports[`<Field /> should match snapshots 2`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -11,8 +11,8 @@
      className=\\"input\\"
      data-field=\\"title\\"
      id=\\"field-title\\"
      onChange={[Function mockConstructor]}
      type=\\"text\\"
-     value=\\"TEST\\"
+     value=\\"\\"
    />
  </div>"
`;
